test/Microsoft.Azure.ServiceBus.UnitTests/API/ApiApprovals.ApproveAzureServiceBus.approved.txt

```diff
@@ -524,4 +527,62 @@ namespace Microsoft.Azure.ServiceBus.InteropExtensions
     {
         public static T GetBody<T>(this Microsoft.Azure.ServiceBus.Message message, System.Runtime.Serialization.XmlObjectSerializer serializer = null) { }
     }
+}
+namespace Microsoft.Azure.ServiceBus.Primitives
+{
+    
+    public class AzureActiveDirectoryTokenProvider : Microsoft.Azure.ServiceBus.Primitives.TokenProvider
+    {
+        public override System.Threading.Tasks.Task<Microsoft.Azure.ServiceBus.Primitives.SecurityToken> GetTokenAsync(string appliesTo, System.TimeSpan timeout) { }
+    }
+    public interface ITokenProvider
+    {
+        System.Threading.Tasks.Task<Microsoft.Azure.ServiceBus.Primitives.SecurityToken> GetTokenAsync(string appliesTo, System.TimeSpan timeout);
+    }
+    public class JsonSecurityToken : Microsoft.Azure.ServiceBus.Primitives.SecurityToken
+    {
+        public JsonSecurityToken(string rawToken, string audience) { }
+    }
+    public class ManagedServiceIdentityTokenProvider : Microsoft.Azure.ServiceBus.Primitives.TokenProvider
+    {
+        public ManagedServiceIdentityTokenProvider() { }
+        public override System.Threading.Tasks.Task<Microsoft.Azure.ServiceBus.Primitives.SecurityToken> GetTokenAsync(string appliesTo, System.TimeSpan timeout) { }
+    }
+    public class SecurityToken
+    {
+        protected string audience;
```

Comment:
Should these fields be protected? They are passed in via based constructor and have public getters to be accessed.

Reply:
expiresAtUtc was only field to be protected and I decided to refactor JsonSecurityToken so that all of them are private now.

---
In reply to: [156761944](https://github.com/Azure/azure-service-bus-dotnet/pull/385#discussion_r156761944) [](ancestors = 156761944)

Reply:
👍 better to have as privates until a need is there for elevated access.